cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -92,6 +92,27 @@ class S3RetryStrategy {
                                                 int64_t attempted_retries) = 0;
 };
 
+/// Wrapper for stock AWS retry strategies
+class AwsRetryStrategy : S3RetryStrategy {
+ public:
+  AwsRetryStrategy(const std::shared_ptr<Aws::Client::RetryStrategy>& retry_strategy) {
+    retry_strategy_ = retry_strategy;
+  }
+  ~AwsRetryStrategy() override;
+
+  bool ShouldRetry(const AWSErrorDetail& error, int64_t attempted_retries) {
+    return retry_strategy_.ShouldRetry(error, attempted_retries);
```

Comment:
I'm not sure an `AWSErrorDetail` will convert into an `AWSError` like this without explicit conversion